pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1031,11 +1031,17 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
 			Sni:              tls.Sni,
 		}
 
-		// If tls.CaCertificate or CaCertificate in Metadata isn't configured don't set up SdsSecretConfig
-		if res.GetRootResourceName() == "" {
+		if res.GetRootResourceName() == "" && features.EnableDefaultToSystemCAValidation.Get() {
+			res = model.SdsCertificateConfig{
+				CaCertificatePath: features.SystemCAFilePath.Get(),
+			}
+		} else {
+			// If tls.CaCertificate or USE_SYSTEM_CA_FALLBACK is not enabled do no validation
 			tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{}
 			tlsContext.Sni = tls.Sni
-		} else {
+		}
```

Comment:
I'm a bit worried about this logic - can you just inside "if res.GetRootName==""" have another if that has the workaround, and leave the rest of the code unchanged. 

Seems a bit confusing.